pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -425,3 +383,86 @@ func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
 	sort.Slice(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
 	return envVars
 }
+
+func gracefulTerminationDurationFromConfig(operatorSpec *operatorv1.StaticPodOperatorSpec) (int, error) {
+	var gracefulTerminationDurationPath = []string{"gracefulTerminationDuration"}
+
+	mergedConfigs, err := resourcemerge.MergeProcessConfig(map[string]resourcemerge.MergeFunc{}, operatorSpec.ObservedConfig.Raw, operatorSpec.UnsupportedConfigOverrides.Raw)
+	if err != nil {
+		return 0, err
+	}
+
+	// read the watch termination from the observed configuration
+	observedConfig := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(mergedConfigs)).Decode(&observedConfig); err != nil {
+		return 0, err
+	}
+	observedGracefulTerminationDurationStr, _, err := unstructured.NestedString(observedConfig, gracefulTerminationDurationPath...)
+	if err != nil {
+		return 0, fmt.Errorf("unable to extract gracefulTerminationDuration from the observed config: %v, path = %v", err, gracefulTerminationDurationPath)
+	}
+	if len(observedGracefulTerminationDurationStr) == 0 {
+		return 0, nil
+	}
+	observedGracefulTerminationDuration, err := strconv.Atoi(observedGracefulTerminationDurationStr)
+	if err != nil {
+		return 0, fmt.Errorf("incorrect value of watchTerminationDuration field in the observed config: %v", err)
+	}
+
+	return observedGracefulTerminationDuration, nil
+}
+
+type kasTemplate struct {
+	Image                       string
+	OperatorImage               string
+	Verbosity                   string
+	GracefulTerminationDuration int
+}
+
+func manageTemplate(rawTemplate string, imagePullSpec string, operatorImagePullSpec string, operatorSpec *operatorv1.StaticPodOperatorSpec) (string, error) {
+	var verbosity string
+	switch operatorSpec.LogLevel {
+	case operatorv1.Normal:
+		verbosity = fmt.Sprintf(" -v=%d", 2)
+	case operatorv1.Debug:
+		verbosity = fmt.Sprintf(" -v=%d", 4)
+	case operatorv1.Trace:
+		verbosity = fmt.Sprintf(" -v=%d", 6)
+	case operatorv1.TraceAll:
+		verbosity = fmt.Sprintf(" -v=%d", 8)
+	default:
+		verbosity = fmt.Sprintf(" -v=%d", 2)
+	}
+
+	gracefulTerminationDuration, err := gracefulTerminationDurationFromConfig(operatorSpec)
+	if err != nil {
+		return "", err
+	}
+
+	if gracefulTerminationDuration == 0 {
+		// apply a default value
+		// 135s is our default value
+		//   the initial 70s is reserved fo the minimal termination period
+		//   additional 60s for finishing all in-flight requests
+		//   an extra 5s to make sure the potential SIGTERM will be sent after the server terminates itself
+		gracefulTerminationDuration = 135
```

Comment:
do we need this? Shouldn't this come from defaultconfig.yaml?

Reply:
yes, we need this.

`defaultconfig.yaml` defines `shutdown-delay-duration`